src/submodule.c

```diff
@@ -149,6 +149,41 @@ static int find_by_path(const git_config_entry *entry, void *payload)
 	return 0;
 }
 
+static int can_add_submodule(git_repository *repo, const char *path) {
+	int error;
+	git_index *index;
+	git_buf dir = GIT_BUF_INIT;
+
+	if ((error = git_buf_sets(&dir, path)) < 0)
+		return error;
+
+	if ((error = git_path_to_dir(&dir)) < 0)
+		return error;
+
+	/* get the index for the repo */
```

Comment:
Would be nice if you removed those comments. You shouldn't describe _what_ you are doing with comments, as most often that is obvious from the code itself. Instead, you should try to document _why_ you are doing that. E.g. a useful comment here would be one preceding the function which explains exactly when you can add a submodule and when you can't.